bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractor.kt

```diff
@@ -0,0 +1,135 @@
+package io.bucketeer.sdk.android.internal.event
+
+import androidx.annotation.VisibleForTesting
+import io.bucketeer.sdk.android.BKTException
+import io.bucketeer.sdk.android.internal.Clock
+import io.bucketeer.sdk.android.internal.IdGenerator
+import io.bucketeer.sdk.android.internal.event.db.EventDao
+import io.bucketeer.sdk.android.internal.logd
+import io.bucketeer.sdk.android.internal.model.Evaluation
+import io.bucketeer.sdk.android.internal.model.Event
+import io.bucketeer.sdk.android.internal.model.User
+import io.bucketeer.sdk.android.internal.remote.ApiClient
+import io.bucketeer.sdk.android.internal.remote.RegisterEventsResult
+import java.util.concurrent.atomic.AtomicReference
+
+internal class EventInteractor(
+  private val eventsMaxBatchQueueCount: Int,
+  private val apiClient: ApiClient,
+  private val eventDao: EventDao,
+  private val clock: Clock,
+  private val idGenerator: IdGenerator,
+) {
+
+  @VisibleForTesting
+  internal val events: AtomicReference<List<Event>> = AtomicReference(emptyList())
+
+  private var eventUpdateListener: EventUpdateListener? = null
+
+  fun setEventUpdateListener(listener: EventUpdateListener?) {
+    this.eventUpdateListener = listener
+  }
+
+  fun trackEvaluationEvent(featureTag: String, user: User, evaluation: Evaluation) {
+    eventDao.addEvent(
+      newEvaluationEvent(clock, idGenerator, featureTag, user, evaluation)
+    )
+
+    updateEventsAndNotify()
+  }
+
+  fun trackDefaultEvaluationEvent(featureTag: String, user: User, featureId: String) {
+    eventDao.addEvent(
+      newDefaultEvaluationEvent(clock, idGenerator, featureTag, user, featureId)
+    )
+
+    updateEventsAndNotify()
+  }
+
+  fun trackGoalEvent(featureTag: String, user: User, goalId: String, value: Double) {
+    eventDao.addEvent(
+      newGoalEvent(clock, idGenerator, goalId, value, featureTag, user)
+    )
+
+    updateEventsAndNotify()
+  }
+
+  fun trackFetchEvaluationsSuccess(
+    featureTag: String,
+    mills: Long,
+    sizeByte: Int,
+  ) {
+    eventDao.addEvents(
+      listOf(
+        newGetEvaluationLatencyMetricsEvent(clock, idGenerator, mills, featureTag),
+        newGetEvaluationSizeMetricsEvent(clock, idGenerator, sizeByte, featureTag)
+      )
+    )
+
+    updateEventsAndNotify()
+  }
+
+  fun trackFetchEvaluationsFailure(
+    featureTag: String,
+    error: BKTException
+  ) {
+    val event = when (error) {
+      is BKTException.NetworkException,
+      is BKTException.TimeoutException -> newTimeoutErrorCountMetricsEvent(
```

Comment:
currently mapping both NetworkException and TimeoutException to `TimeoutErrorCountMetricsEvent`

Reply:
I'm thinking of adding a `NetworkErrorCountMetricsEvent` to report it separately.
I will confirm and return to you next week, but for now, it is okay.